Jellyfin.Server/Program.cs

```diff
@@ -664,5 +659,49 @@ private static string NormalizeCommandLineArgument(string arg)
 
             return "\"" + arg + "\"";
         }
+
+        private static string GetUnixSocketPath(IConfiguration startupConfig, IApplicationPaths appPaths)
+        {
+            var socketPath = startupConfig.GetUnixSocketPath();
+
+            if (string.IsNullOrEmpty(socketPath))
+            {
+                var xdgRuntimeDir = Environment.GetEnvironmentVariable("XDG_RUNTIME_DIR");
+                var socketFile = "jellyfin.sock";
+                if (xdgRuntimeDir == null)
+                {
+                    // Fall back to config dir
+                    socketPath = Path.Join(appPaths.ConfigurationDirectoryPath, socketFile);
+                }
+                else
+                {
+                    socketPath = Path.Join(xdgRuntimeDir, socketFile);
+                }
+            }
+
+            return socketPath;
+        }
+
+        private static void SetUnixSocketPermissions(IConfiguration startupConfig, string socketPath)
+        {
+            var socketPerms = startupConfig.GetUnixSocketPermissions();
+
+            if (!string.IsNullOrEmpty(socketPerms))
+            {
+                [DllImport("libc")]
```

Comment:
I think a dll import for a one-off during startup is sub-optimal. Can't we just call chmod directly?

Reply:
Is the chmod cross-platform? I'm not sure if any of this is, but that could be why? @knackebrot

Reply:
I thought P/Invoke would be more efficient than spawning a new process and a safer bet, since .net runtime is linked against libc. I don't know much about how P/Invoke works internally, can you please explain how is it sub-optimal?